frontend/src/components/Layout/VirtualGrid/pipeline.ts

```diff
@@ -73,9 +73,9 @@ export function getScrollParents(
     ? /(auto|scroll|hidden)/
     : /(auto|scroll)/;
 
-  let vertical: HTMLElement | undefined;
-  let horizontal: HTMLElement | undefined;
-  let parent: HTMLElement | null = element;
+  let vertical: HTMLElement | SVGElement | undefined;
+  let horizontal: HTMLElement | SVGElement | undefined;
+  let parent: HTMLElement | SVGElement | null = element;
```

Comment:
Why?

Reply:
Honestly, I don't think this makes sense just for the sake of removing ``as``s. TypeScript is a way to assist in the type inference of stuff in the JavaScript world, and ``as``, although not ideal, is indeed a hint to help it in some occasions do the correct inference. Sometimes even ``any`` is necessary to do more complex inferences like [here](https://github.com/jellyfin/jellyfin-vue/blob/vite/frontend/src/composables/use-datefns.ts#L38) (and I looked a lot of libraries to see how similar typings were accomplished, with all of them using ``any``).

At no point is logical to find a CSS virtual grid inside an svg. It's silly to assume we could be dealing with this at runtime, because it's illogical to do so. And this just adds complexity (dealing with SVGElement typings) just for the sake of making the "linter" happy, in my opinion.

@ThibaultNocchi any opinions?

Reply:
The other option would be to do add an `&& el instanceof HTMLElement` to the `if` statement on line 114. This would accomplish the same thing by only allowing an HTML element into this function. Happy to make that change if you think it's better.

I do think `as` casts have a use, but they should be used as an escape hatch rather than a convenience. I don't think that applies here since it's pretty simple to narrow the types before calling the function or expand the types of the function parameter to avoid needing one.

Reply:
@aweebs Sounds good to me!